pbs_light.go

```diff
@@ -440,6 +403,66 @@ func auction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 	mRequestTimer.UpdateSince(pbs_req.Start)
 }
 
+// sortBidsAddKeywordsMobile sorts the bids and adds ad server targeting keywords to each bid.
+// The bids are sorted by cpm to find the highest bid.
+// The ad server targeting keywords are added to all bids, with specific keywords for the highest bid.
+func sortBidsAddKeywordsMobile(bids pbs.PBSBidSlice, pbs_req *pbs.PBSRequest, priceGranularitySetting string) {
+	if priceGranularitySetting == "" {
+		priceGranularitySetting = defaultPriceGranularity
+	}
+
+	// record bids by ad unit code for sorting
+	code_bids := make(map[string]pbs.PBSBidSlice, len(bids))
+	for _, bid := range bids {
+		code_bids[bid.AdUnitCode] = append(code_bids[bid.AdUnitCode], bid)
+	}
+
+	// loop through ad units to find top bid
+	for _, unit := range pbs_req.AdUnits {
+		bar := code_bids[unit.Code]
```

Comment:
why is this called `bar` ?

Reply:
probly written by people with too little time to spare...

she's just moving this block of code, though--not writing it from scratch... so i don't think it should hold up the merge.